Subprocess: increase number of threads and return to blocking pipes.

penguinolog opened the pull request:
Found several use-cases, when non-blocking pipes can cause data loose.

In tests not checking mised stdout/stderr because of expected
race condition during log output.

- [x] I think the code is well written
- [x] Unit tests for the changes exist

rayburgemeestre reviewed (approved):
Really like how the code is simplified now

penguinolog committed:
Subprocess: increase number of threads and return to blocking pipes.

Found several use-cases, when non-blocking pipes can cause data loose.

In tests not checking mised stdout/stderr because of expected
race condition during log output.

Signed-off-by: Alexey Stepanov <penguinolog@gmail.com>
